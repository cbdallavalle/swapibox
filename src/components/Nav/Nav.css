 @font-face { font-family: deathStar; src: url('../../styles/deathStar.otf'); } 


 a {
  display: block;
  height: 20px;
  font-family: 'Dhurjati', sans-serif;
  text-decoration: none;
  font-size: 23px;
  color: white;
 }

 a h1 {
  margin: 20px auto 40px auto;
  padding: 6px;
  width: 300px;
  background-color: #ffffff3b;
  border-radius: 20px;
  color: white;
  font-family: deathStar, sans-serif;
  font-size: 25px;
 }

 button {
  padding: 6px 25px 24px 25px;
  background-color: #ffffff3b;
  border: 2px solid white;
  border-left: none;
 }

 button:hover,
 button:active,
 button:focus {
  background-color: #ffffffad;
  outline: none;
 }

 nav {
  height: 290px;
  background-image: url('../../styles/space.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: bottom;
}

.nav-btns-cont {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 25px auto;
  width: 100%;
  height: 50px;
}

.nav-img {
  height: 40px;
}

.title-cont {
  height: 180px;
}

#death-star {
  position: absolute;
  top: 35px;
  right: 50px;
  width: 200px;
}

#empire-logo {
  margin-top: 20px;
  width: 60px;
}

#people {
  border-left: 2px solid white;
  border-top-left-radius: 15px;
  border-bottom-left-radius: 15px;
}

#targets {
  border-top-right-radius: 15px;
  border-bottom-right-radius: 15px;
}

@media only screen and (max-width: 380px) {
  .nav-img {
    height: 30px;
  }

  button {
    padding: 9px 20px 20px 20px;
  }
}

@media only screen and (max-width: 853px) {
  #death-star {
    display: none;
  }
}
