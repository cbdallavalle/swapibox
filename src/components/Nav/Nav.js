import React from 'react';
import PropTypes from 'prop-types';
import { NavLink, Link } from 'react-router-dom';
import empireImg from '../../styles/GalacticEmpire.png';
import people from '../../styles/people.svg';
import planet from '../../styles/earth-globe.svg';
import vehicle from '../../styles/star-wars.svg';
import target from '../../styles/bullseye.svg';
import deathStar from '../../styles/deathStar.png';
import './Nav.css';

const Nav = ({ renderCards, getFilmCrawl }) => {
  return (
    <nav>
      <div className="title-cont">
        <Link 
          to='/'
          onClick={ getFilmCrawl }
        >
          <img src={empireImg} alt="empire logo" id="empire-logo" />
          <h1>The Imperial Almanac</h1>
        </Link>
      </div>
      <div className="nav-btns-cont">
        <NavLink to='/people'>
          <button 
            id="people"
            onClick={ () => renderCards("people") }
          >
            <img className="nav-img" src={ people } alt="search people" />
          </button>
        </NavLink>
        <NavLink to='/planets'>
          <button onClick={ () => renderCards("planets") }>
            <img className="nav-img" src={ planet } alt="search people" />
          </button>
        </NavLink>
        <NavLink to='/vehicles'>
          <button onClick={ () => renderCards("vehicles") }>
            <img className="nav-img" src={ vehicle } alt="search people" />
          </button>
        </NavLink>
        <NavLink to='/targets'>
          <button id="targets"
          >
            <img className="nav-img" src={ target } alt="search people" />
          </button>
        </NavLink>
      </div>
      <img id="death-star" src={ deathStar } />
    </nav>
  );
};

const { func } = PropTypes;

Nav.propTypes = {
  renderCards: func.isRequired,
  getFilmCrawl: func.isRequired
};

export default Nav;