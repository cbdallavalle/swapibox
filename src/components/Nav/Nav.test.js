/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav.js';

describe('Nav', () => {
  let wrapper;
  const mockRenderCards = jest.fn();
  const mockGetFilmCrawl = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Nav 
        renderCards={ mockRenderCards } 
        getFilmCrawl={ mockGetFilmCrawl }
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should trigger renderCards when clicked', () => {
    wrapper.find('#people').first().simulate('click');
    expect(mockRenderCards).toHaveBeenCalled();
  })

  it('should trigger getFilmCrawl when Logo clicked', () => {
    wrapper.find('Link').simulate('click');
    expect(mockGetFilmCrawl).toHaveBeenCalled();
  })
})