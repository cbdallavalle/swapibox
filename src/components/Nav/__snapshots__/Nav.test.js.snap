// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    getFilmCrawl={[Function]}
    renderCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title-cont"
>
          <Link
                    onClick={[Function]}
                    replace={false}
                    to="/"
          >
                    <img
                              alt="empire logo"
                              id="empire-logo"
                              src="GalacticEmpire.png"
                    />
                    <h1>
                              The Imperial Almanac
                    </h1>
          </Link>
</div>,
        <div
          className="nav-btns-cont"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/people"
          >
                    <button
                              id="people"
                              onClick={[Function]}
                    >
                              <img
                                        alt="search people"
                                        className="nav-img"
                                        src="people.svg"
                              />
                    </button>
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/planets"
          >
                    <button
                              onClick={[Function]}
                    >
                              <img
                                        alt="search people"
                                        className="nav-img"
                                        src="earth-globe.svg"
                              />
                    </button>
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/vehicles"
          >
                    <button
                              onClick={[Function]}
                    >
                              <img
                                        alt="search people"
                                        className="nav-img"
                                        src="star-wars.svg"
                              />
                    </button>
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/targets"
          >
                    <button
                              id="targets"
                    >
                              <img
                                        alt="search people"
                                        className="nav-img"
                                        src="bullseye.svg"
                              />
                    </button>
          </NavLink>
</div>,
        <img
          id="death-star"
          src="deathStar.png"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            onClick={[Function]}
            replace={false}
            to="/"
>
            <img
                        alt="empire logo"
                        id="empire-logo"
                        src="GalacticEmpire.png"
            />
            <h1>
                        The Imperial Almanac
            </h1>
</Link>,
          "className": "title-cont",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="empire logo"
                id="empire-logo"
                src="GalacticEmpire.png"
/>,
              <h1>
                The Imperial Almanac
</h1>,
            ],
            "onClick": [Function],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "empire logo",
                "id": "empire-logo",
                "src": "GalacticEmpire.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The Imperial Almanac",
              },
              "ref": null,
              "rendered": "The Imperial Almanac",
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/people"
>
              <button
                            id="people"
                            onClick={[Function]}
              >
                            <img
                                          alt="search people"
                                          className="nav-img"
                                          src="people.svg"
                            />
              </button>
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/planets"
>
              <button
                            onClick={[Function]}
              >
                            <img
                                          alt="search people"
                                          className="nav-img"
                                          src="earth-globe.svg"
                            />
              </button>
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/vehicles"
>
              <button
                            onClick={[Function]}
              >
                            <img
                                          alt="search people"
                                          className="nav-img"
                                          src="star-wars.svg"
                            />
              </button>
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/targets"
>
              <button
                            id="targets"
              >
                            <img
                                          alt="search people"
                                          className="nav-img"
                                          src="bullseye.svg"
                            />
              </button>
</NavLink>,
          ],
          "className": "nav-btns-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                id="people"
                onClick={[Function]}
>
                <img
                                alt="search people"
                                className="nav-img"
                                src="people.svg"
                />
</button>,
              "to": "/people",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="search people"
                  className="nav-img"
                  src="people.svg"
/>,
                "id": "people",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "search people",
                  "className": "nav-img",
                  "src": "people.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                onClick={[Function]}
>
                <img
                                alt="search people"
                                className="nav-img"
                                src="earth-globe.svg"
                />
</button>,
              "to": "/planets",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="search people"
                  className="nav-img"
                  src="earth-globe.svg"
/>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "search people",
                  "className": "nav-img",
                  "src": "earth-globe.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                onClick={[Function]}
>
                <img
                                alt="search people"
                                className="nav-img"
                                src="star-wars.svg"
                />
</button>,
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="search people"
                  className="nav-img"
                  src="star-wars.svg"
/>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "search people",
                  "className": "nav-img",
                  "src": "star-wars.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                id="targets"
>
                <img
                                alt="search people"
                                className="nav-img"
                                src="bullseye.svg"
                />
</button>,
              "to": "/targets",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="search people"
                  className="nav-img"
                  src="bullseye.svg"
/>,
                "id": "targets",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "search people",
                  "className": "nav-img",
                  "src": "bullseye.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "death-star",
          "src": "deathStar.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title-cont"
>
            <Link
                        onClick={[Function]}
                        replace={false}
                        to="/"
            >
                        <img
                                    alt="empire logo"
                                    id="empire-logo"
                                    src="GalacticEmpire.png"
                        />
                        <h1>
                                    The Imperial Almanac
                        </h1>
            </Link>
</div>,
          <div
            className="nav-btns-cont"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/people"
            >
                        <button
                                    id="people"
                                    onClick={[Function]}
                        >
                                    <img
                                                alt="search people"
                                                className="nav-img"
                                                src="people.svg"
                                    />
                        </button>
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/planets"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    <img
                                                alt="search people"
                                                className="nav-img"
                                                src="earth-globe.svg"
                                    />
                        </button>
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/vehicles"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    <img
                                                alt="search people"
                                                className="nav-img"
                                                src="star-wars.svg"
                                    />
                        </button>
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/targets"
            >
                        <button
                                    id="targets"
                        >
                                    <img
                                                alt="search people"
                                                className="nav-img"
                                                src="bullseye.svg"
                                    />
                        </button>
            </NavLink>
</div>,
          <img
            id="death-star"
            src="deathStar.png"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              onClick={[Function]}
              replace={false}
              to="/"
>
              <img
                            alt="empire logo"
                            id="empire-logo"
                            src="GalacticEmpire.png"
              />
              <h1>
                            The Imperial Almanac
              </h1>
</Link>,
            "className": "title-cont",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt="empire logo"
                  id="empire-logo"
                  src="GalacticEmpire.png"
/>,
                <h1>
                  The Imperial Almanac
</h1>,
              ],
              "onClick": [Function],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "empire logo",
                  "id": "empire-logo",
                  "src": "GalacticEmpire.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "The Imperial Almanac",
                },
                "ref": null,
                "rendered": "The Imperial Almanac",
                "type": "h1",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/people"
>
                <button
                                id="people"
                                onClick={[Function]}
                >
                                <img
                                                alt="search people"
                                                className="nav-img"
                                                src="people.svg"
                                />
                </button>
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/planets"
>
                <button
                                onClick={[Function]}
                >
                                <img
                                                alt="search people"
                                                className="nav-img"
                                                src="earth-globe.svg"
                                />
                </button>
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/vehicles"
>
                <button
                                onClick={[Function]}
                >
                                <img
                                                alt="search people"
                                                className="nav-img"
                                                src="star-wars.svg"
                                />
                </button>
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/targets"
>
                <button
                                id="targets"
                >
                                <img
                                                alt="search people"
                                                className="nav-img"
                                                src="bullseye.svg"
                                />
                </button>
</NavLink>,
            ],
            "className": "nav-btns-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  id="people"
                  onClick={[Function]}
>
                  <img
                                    alt="search people"
                                    className="nav-img"
                                    src="people.svg"
                  />
</button>,
                "to": "/people",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="search people"
                    className="nav-img"
                    src="people.svg"
/>,
                  "id": "people",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "search people",
                    "className": "nav-img",
                    "src": "people.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  onClick={[Function]}
>
                  <img
                                    alt="search people"
                                    className="nav-img"
                                    src="earth-globe.svg"
                  />
</button>,
                "to": "/planets",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="search people"
                    className="nav-img"
                    src="earth-globe.svg"
/>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "search people",
                    "className": "nav-img",
                    "src": "earth-globe.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  onClick={[Function]}
>
                  <img
                                    alt="search people"
                                    className="nav-img"
                                    src="star-wars.svg"
                  />
</button>,
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="search people"
                    className="nav-img"
                    src="star-wars.svg"
/>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "search people",
                    "className": "nav-img",
                    "src": "star-wars.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  id="targets"
>
                  <img
                                    alt="search people"
                                    className="nav-img"
                                    src="bullseye.svg"
                  />
</button>,
                "to": "/targets",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="search people"
                    className="nav-img"
                    src="bullseye.svg"
/>,
                  "id": "targets",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "search people",
                    "className": "nav-img",
                    "src": "bullseye.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "death-star",
            "src": "deathStar.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
